File: backend/app/core/upgrades/curses/health_reduc_5.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class HealthReduc5(UpgradeImplementation):
    id = 'health_reduc_5'
    name = 'Reduced Health 5'
    description = '-80 max health.'
    rarity = 'legendary'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -80
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/health_reduc_5.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const HealthReduc5Def: UpgradeDef = {
  id: "health_reduc_5",
  name: "Reduced Health 5",
  description: "-80 max health.",
  rarity: RarityID.Legendary,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: -80,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class HealthReduc5 extends Upgrade {}
